pkg/store/bucket_e2e_test.go

```diff
@@ -215,6 +215,31 @@ func TestBucketStore_e2e(t *testing.T) {
 			MaxTime: maxt,
 		}, srv))
 		testutil.Equals(t, 0, len(srv.SeriesSet))
+
+		// Test the samples limit.
+		testutil.Ok(t, os.RemoveAll(dir))
+		s = prepareStoreWithTestBlocks(t, dir, bkt, 30)
+		mint, maxt = s.store.TimeRange()
+		defer s.Close()
+
+		srv = newStoreSeriesServer(ctx)
+
+		testutil.Ok(t, s.store.Series(&storepb.SeriesRequest{
+			Matchers: []storepb.LabelMatcher{
+				{Type: storepb.LabelMatcher_EQ, Name: "a", Value: "1"},
+			},
+			MinTime: mint,
+			MaxTime: maxt,
+		}, srv))
+
+		testutil.NotOk(t, s.store.Series(&storepb.SeriesRequest{
+			Matchers: []storepb.LabelMatcher{
+				{Type: storepb.LabelMatcher_RE, Name: "a", Value: "1|2"},
+			},
+			MinTime: mint,
+			MaxTime: maxt,
+		}, srv))
+
```

Comment:
Would be great to have test for the `queryGate` also but not sure how to do this.
Would need one blocking query and second one checked for a timeout? Still this would be fragile I suppose :/

Reply:
The gate is super simple so not sure